### ISSUE TITLE
[rush] Doesn't use node installed by volta

### ISSUE THREAD

**Issue by knownasilya** [microsoft/rushstack#1997]
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**

Install volta: https://volta.sh/

Install a specific version of node: `volta install node@10.13.0`
Try `rush install`, it says that another version of node is being used.
Although `node -v` returns the correct version.
`which node` -> `/Users/ilya/.volta/bin/node`
Rush uses: `"/Users/ilya/.rush/node-v14.5.0/pnpm-4.12.5"`

Using volta 0.6.5

**What is the expected behavior?**

Expect rush to use the version of node that volta installs.

* **Tool: rush** 
* **Tool Version: 5.27.3** 
* **OS: macos latest**

**Comment by iclanton** [comment]
I can repro this.

This seems to be a bug in Volta. I just replaced Rush's `start.js` script with the simple `console.log(process.execPath);` and in a Windows sandbox that prints `C:\Users\WDAGUtilityAccount\AppData\Local\Volta\tools\image\node\12.18.2\node.exe`.

The file at `C:\Users\WDAGUtilityAccount\AppData\Local\Volta\tools\user\bins\rush.json`, which appears to be a manifest file Volta uses for the `rush` bin looks like this:

```JSON
{
  "name": "rush",
  "package": "@microsoft/rush",
  "version": "5.27.3",
  "path": "./bin/rush",
  "platform": {
    "node": {
      "runtime": "12.18.2",
      "npm": null
    },
    "yarn": null
  },
  "loader": {
    "command": "node",
    "args": []
  }
}
```

And on my machine, that specifically references Node v12.18.2, even though I ran `volta install node@10.13.0` and `node -v` prints out `10.13.0`.
